airflow/providers/fab/auth_manager/security_manager/override.py

```diff
@@ -276,6 +276,7 @@ class FabAirflowSecurityManagerOverride(AirflowSecurityManagerV2):
         (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_VARIABLE),
         (permissions.ACTION_CAN_DELETE, permissions.RESOURCE_VARIABLE),
         (permissions.ACTION_CAN_DELETE, permissions.RESOURCE_XCOM),
+        (permissions.ACTION_CAN_DELETE, permissions.RESOURCE_DATASET),
```

Comment:
Hmm, does this mean a FAB provider upgrade would be necessary for the endpoints to work?Do we need to do something about this? Documentation? No quite sure how this works. cc @vincbeck

Reply:
> does this mean a FAB provider upgrade would be necessary for the endpoints to work?

I think so

> Do we need to do something about this?

Without this, `@security.requires_access_dataset("DELETE")` won't work

Reply:
Technically the endpoint will work but users wont have permissions. 100% on what @uranusjr said [here](https://github.com/apache/airflow/pull/37176#issuecomment-1951803492)